### PR TITLE
Closes 120 - Replace metacore with metadata

### PR DESCRIPTION
**Pull request by vedhav**
### Thank you for your Pull Request! 

We have developed a Pull Request template to aid you and our reviewers.  Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the xportr codebase remains robust and consistent.  

### The scope of `{xportr}`

`{xportr}`'s scope is to enable R users to write out submission compliant `xpt` files that can be delivered to a Health Authority or to downstream validation software programs.  We see labels, lengths, types, ordering and formats from a dataset specification object (SDTM and ADaM) as being our primary focus. We also see messaging and warnings to users around applying information from the specification file as a primary focus. Please make sure your Pull Request meets this **scope of {xportr}**. If your Pull Request moves beyond this scope, please get in touch with the `{xportr}` team on [slack](https://pharmaverse.slack.com/archives/C030EB2M4GM) or create an issue to discuss.

Please check off each task box as an acknowledgment that you completed the task. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `devel` branch until you have checked off each task.

### Changes Description

1. Replaces the `metacore` with `metadata` across all the xportr functions
2. Provide deprecation message for the users
3. Set up deprecation badges for future deprecation by running `usethis::use_lifecycle()` 

### Task List

- [x] The spirit of xportr is met in your Pull Request
- [x] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [x] Summary of changes filled out in the above Changes Description.  Can be removed or left blank if changes are minor/self-explanatory.
- [x] Check that your Pull Request is targeting the `devel` branch, Pull Requests to `main` should use the [Release Pull Request Template](https://github.com/atorus-research/xportr/tree/94_pr_template/.github/PULL_REQUEST_TEMPLATE)
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/).  Use `styler` package and functions to style files accordingly.
- [x] Updated relevant unit tests or have written new unit tests. See our [Wiki](https://github.com/atorus-research/xportr/wiki/Style-Guide-for-Unit-Tests) for conventions used in this package.
- [x] Creation/updated relevant roxygen headers and examples. See our [Wiki](https://github.com/atorus-research/xportr/wiki/Style-Guide-for-Roxygen-Headers) for conventions used in this package.
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Run `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new/updated functions occur on the "Reference" page.
- [x] Update NEWS.md if the changes pertain to a user-facing function (i.e. it has an @export tag) or documentation aimed at users (rather than developers)
- [x] Address any updates needed for vignettes and/or templates
- [x] Link the issue Development Panel so that it closes after successful merging.
- [x] Fix merge conflicts 
- [x] Pat yourself on the back for a job well done!  Much love to your accomplishment!

### CODE REVIEW

File: NEWS.md [hunk 1]
```diff
@@ -12,6 +12,9 @@
 
 * Moved `{pkgdown}` site to bootswatch. Enabled search and linked slack icon (#122).
 
+## Deprecation and Breaking Changes
+
+* The `metacore` argument has been renamed to `metadata` in the following six xportr functions: `xportr_df_label`, `xportr_format`, `xportr_label`, `xportr_length`, `xportr_order`, and `xportr_type`. Please update your code to use the new `metadata` argument in place of `metacore`.
```
Comment (bms63) [inline NEWS.md:17]:
```suggestion
* The `metacore` argument has been renamed to `metadata` in the following six xportr functions: `xportr_df_label()`, `xportr_format()`, `xportr_label()`, `xportr_length()`, `xportr_order()`, and `xportr_type()`. Please update your code to use the new `metadata` argument in place of `metacore`.
```

I think they become linkable on the website with the `()`

File: R/format.R [hunk 2]
```diff
@@ -20,14 +21,27 @@
...
+    .df,
+    metadata,
+    domain = NULL,
+    verbose = getOption("xportr.length_verbose", "none"),
+    metacore = deprecated()) {
+  if (!missing(metacore)) {
+    lifecycle::deprecate_warn(
+      when = "0.0.3",
+      what = "xportr_format(metacore = )",
+      with = "xportr_format(metadata = )"
```
Comment (bms63) [inline R/format.R:41]:
```suggestion
    lifecycle::deprecate_warn(
      when = "0.3.0",
      what = "xportr_format(metacore = )",
      with = "xportr_format(metadata = )"
```

File: R/label.R [hunk 3]
```diff
@@ -22,15 +23,27 @@
...
+    .df,
+    metadata,
+    domain = NULL,
+    verbose = getOption("xportr.length_verbose", "none"),
+    metacore = deprecated()) {
+  if (!missing(metacore)) {
+    lifecycle::deprecate_warn(
+      when = "0.0.3",
+      what = "xportr_format(metacore = )",
+      with = "xportr_format(metadata = )"
```
Comment (bms63) [inline R/label.R:43]:
```suggestion
      when = "0.3.0",
      what = "xportr_format(metacore = )",
      with = "xportr_format(metadata = )"
```

File: R/type.R [hunk 4]
```diff
@@ -30,9 +31,21 @@
...
+    .df,
+    metadata,
+    domain = NULL,
+    verbose = getOption("xportr.length_verbose", "none"),
+    metacore = deprecated()) {
+  if (!missing(metacore)) {
+    lifecycle::deprecate_warn(
+      when = "0.0.3",
+      what = "xportr_format(metacore = )",
+      with = "xportr_format(metadata = )"
```
Comment (bms63) [inline R/type.R:45]:
```suggestion
    lifecycle::deprecate_warn(
      when = "0.3.0",
      what = "xportr_format(metacore = )",
      with = "xportr_format(metadata = )"
```

**Comment by bms63** [comment]
Is it possible to drop the `docs` folder or merge in Andre's removal of `docs`.  Looks like the link checker is falling over from the docs as well as the merge conflicts

**Review by elimillera — APPROVED** [review]
LGTM

**Pull request merged**